scripts/menu/server/sv_player_modal.lua

```diff
@@ -68,9 +68,25 @@ RegisterNetEvent('txAdmin:menu:spectatePlayer', function(id)
     if not target then
       return
     end
+    local tgtBucket = GetPlayerRoutingBucket(id)
+    local srcBucket = GetPlayerRoutingBucket(src)
+    if tgtBucket ~= srcBucket then 
+      ADMIN_DATA[tostring(src)].bucket = srcBucket
+      SetPlayerRoutingBucket(src, tgtBucket)
+    end
 
     local tgtCoords = GetEntityCoords(target)
     TriggerClientEvent('txAdmin:menu:specPlayerResp', src, id, tgtCoords)
   end
   TriggerEvent('txaLogger:menuEvent', src, 'spectatePlayer', allow, id)
+end)
+
+
+
+RegisterNetEvent('txAdmin:menu:endSpectate', function()
```

Comment:
This net event can be used to spam errors or change an admins bucket without any verification. Please run verification checks on the calling source using the HasPermission function and `players.spectate`

Reply:
Will do my bad